packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -857,32 +886,10 @@ export const Table = <
     ) => {
       const row = page[index];
       prepareRow(row);
+      const isRowASubComponent =
+        !!row.original[iuiId as keyof typeof row.original];
 
-      if (row.subRows.length > 0 || !subComponent) {
-        return (
-          <TableRowMemoized
-            row={row}
-            rowProps={rowProps}
-            isLast={index === page.length - 1}
-            onRowInViewport={onRowInViewportRef}
-            onBottomReached={onBottomReachedRef}
-            intersectionMargin={intersectionMargin}
-            state={state}
-            key={row.getRowProps().key}
-            onClick={onRowClickHandler}
-            subComponent={subComponent}
-            isDisabled={!!isRowDisabled?.(row.original)}
-            tableHasSubRows={hasAnySubRows}
-            tableInstance={instance}
-            expanderCell={expanderCell}
-            scrollContainerRef={tableRef.current}
-            tableRowRef={enableVirtualization ? undefined : tableRowRef(row)}
-            density={density}
-            virtualItem={virtualItem}
-            virtualizer={virtualizer}
-          />
-        );
-      } else {
+      if (isRowASubComponent && !!subComponent) {
```

Comment:
This might be a nit but i think the `!!subComponent` check can be moved up with `isRowASubComponent`.

```js
      const isRowASubComponent =
        !!row.original[iuiId as keyof typeof row.original] && !!subComponent;

      if (isRowASubComponent) {
```

Reply:
Also, I don't think we need to set `as keyof typeof row.original`. It's basically lying to TS.

It would be better to just use `as any`.

Reply:
I have made the change.

Reply:
I think you missed the [first comment](https://github.com/iTwin/iTwinUI/pull/2306#discussion_r1806575479). I'm still seeing `!!subComponent` in its original place.

![](https://github.com/user-attachments/assets/d4c33ca8-46e6-40a3-9f70-2dfba49d3ffb)

Reply:
You're right, I missed the first comment. I have made the change.

Reply:
Now you've undone the second comment, where I suggested using `as any`.

The correct snippet should be:

```ts
      const isRowASubComponent = !!row.original[iuiId as any] && !!subComponent;

      if (isRowASubComponent) {
```

Reply:
I apologize for now reviewing this carefully. I have made the change.